src/flatpakTerminal.ts

```diff
@@ -1,113 +1,57 @@
 import * as vscode from 'vscode'
-import * as child_process from 'child_process'
-import * as readline from 'readline'
-import { failure, finished, newTask } from './store'
-import { TaskMode } from './taskMode'
-import { Command } from './command'
 
 export class FlatpakTerminal {
-  private commands: Command[] = []
-  private currentCommand: number
-  public failed: boolean
-  private isRunning = false
-  private mode?: TaskMode
-  private output: vscode.OutputChannel
-  private currentProcess?: child_process.ChildProcessWithoutNullStreams
-  public completeBuild = false
-
-  constructor(outputChannel: vscode.OutputChannel) {
-    this.currentCommand = 0
-    this.output = outputChannel
-    this.failed = false
-  }
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  close(): void {
-    if (this.currentProcess) {
-      this.currentProcess.emit('close', -1)
+    private inner?: vscode.Terminal
+    private pty: vscode.Pseudoterminal
+    private emitter: vscode.EventEmitter<string>
+
+    private readonly _onDidClose = new vscode.EventEmitter<void>()
+    readonly onDidClose = this._onDidClose.event
+
+    constructor() {
+        this.emitter = new vscode.EventEmitter<string>();
+        this.pty = {
+            open: () => console.log("Flatpak terminal opened"),
+            close: () => {
+                this._onDidClose.fire()
+                this.inner?.dispose()
+                this.inner = undefined
+            },
+            onDidWrite: this.emitter.event,
+        }
     }
-  }
 
-  async setCommands(commands: Command[], mode: TaskMode): Promise<void> {
-    if (this.isRunning) {
-      this.commands = [...this.commands, ...commands]
-    } else {
-      this.commands = commands
-      this.mode = mode
-      this.currentCommand = 0
-      this.failed = false
-      await this.spawn(this.commands[0])
+    append(content: string): void {
+        this.emitter.fire(content)
     }
-  }
 
-  onError(message: string, command: Command): void {
-    this.writeError(message)
-    failure({ command, message })
-    this.failed = true
-    this.isRunning = false
-  }
+    appendMessage(message: string, isErr: boolean): void {
+        const boldRed = '\x1b[1;31m'
+        const boldWhite = '\x1b[1;37m'
```

Comment:
Why bold by default?

Reply:
Well I am not sure whether this is used only for displaying the executed command or is it used for all the outputs that are not errors

Reply:
appendMessage is used for the executed command and errors. The output of the process is directed to it untampered

Reply:
`Nothing to do` also uses appendMessage. What about `appendError` and `appendMessage`?

Reply:
Sounds good